### PR TITLE
angular: improve form service

### PR DESCRIPTION
**Pull request by mshima**
Simplify Id FromControls:
- Id FormControls are disabled, they are not editable.
- Rework types definitions in preparation for Angular 14.

Introduce FormDefaults type to be verbose on fields that are non nullable at new entities due to default value.

Related to https://github.com/jhipster/generator-jhipster/issues/18817

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.spec.ts.ejs [hunk 1]
```diff
@@ -82,37 +82,50 @@ _%>
 
   describe('Service methods', () => {
     describe('create<%= entityAngularName %>FormGroup', () => {
-      it('should create a new form without <%= primaryKey.name %> FromControl', () => {
+      it('should create a new form with form controls', () => {
```
Comment (qmonmert) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.spec.ts.ejs:85]: maybe `form controls` => `FormControl`

**Review by qmonmert — DISMISSED** [review]
LGTM 👏

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.spec.ts.ejs (hunk 1 above)
Comment (mshima) [reply generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-form.service.spec.ts.ejs:85]:
```suggestion
      it('should create a new form with FormControl', () => {
```

**Comment by mshima** [comment]
Thanks @qmonmert

**Pull request merged**